// stylelint/dist/rules/declaration-block-trailing-semicolon/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = declarationBlockTrailingSemicolon;
var ruleName = "declaration-block-trailing-semicolon";

exports.ruleName = ruleName;
var messages = {
  expected: "Expected a trailing semicolon (" + ruleName + ")",
  rejected: "Unexpected trailing semicolon (" + ruleName + ")" };

exports.messages = messages;
/**
 * @param {"always"|"never"} options
 */

function declarationBlockTrailingSemicolon(options) {
  return function (css, result) {
    var checkRule = undefined;
    if (options === "always") {
      checkRule = expectSemicolon;
    } else if (options === "never") {
      checkRule = rejectSemicolon;
    } else {
      return;
    }

    css.eachRule(checkRule);

    function expectSemicolon(rule) {
      if (rule.semicolon) {
        return;
      }

      result.warn(messages.expected, { node: rule });
    }

    function rejectSemicolon(rule) {
      if (!rule.semicolon) {
        return;
      }

      result.warn(messages.rejected, { node: rule });
    }
  };
}
