// stylelint/dist/rules/comment-space-inside/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _utils = require("../../utils");

var ruleName = "comment-space-inside";

exports.ruleName = ruleName;
var messages = (0, _utils.ruleMessages)(ruleName, {
  expectedOpening: "Expected single space after \"/*\"",
  rejectedOpening: "Unexpected whitespace after \"/*\"",
  expectedClosing: "Expected single space before \"*/\"",
  rejectedClosing: "Unexpected whitespace before \"*/\""
});

exports.messages = messages;

exports["default"] = function (expectation) {
  return function (root, result) {
    var validOptions = (0, _utils.validateOptions)(result, ruleName, {
      actual: expectation,
      possible: ["always", "never"]
    });
    if (!validOptions) {
      return;
    }

    root.walkComments(function (comment) {

      if (comment.raws.inline) {
        return;
      }

      var rawComment = comment.toString();
      var leftMatches = rawComment.match(/(^\/\*+)(\s)?/);
      var rightMatches = rawComment.match(/(\s)?(\*+\/)$/);
      var opener = leftMatches[1];
      var leftSpace = leftMatches[2] || "";
      var rightSpace = rightMatches[1] || "";
      var closer = rightMatches[2];

      if (expectation === "never" && leftSpace !== "") {
        (0, _utils.report)({
          message: messages.rejectedOpening,
          node: comment,
          index: opener.length,
          result: result,
          ruleName: ruleName
        });
      }
      if (expectation === "always" && !(0, _utils.isWhitespace)(leftSpace)) {
        (0, _utils.report)({
          message: messages.expectedOpening,
          node: comment,
          index: opener.length,
          result: result,
          ruleName: ruleName
        });
      }

      if (expectation === "never" && rightSpace !== "") {
        (0, _utils.report)({
          message: messages.rejectedClosing,
          node: comment,
          index: comment.toString().length - closer.length - 1,
          result: result,
          ruleName: ruleName
        });
      }
      if (expectation === "always" && !(0, _utils.isWhitespace)(rightSpace)) {
        (0, _utils.report)({
          message: messages.expectedClosing,
          node: comment,
          index: comment.toString().length - closer.length - 1,
          result: result,
          ruleName: ruleName
        });
      }
    });
  };
};
