// stylelint/dist/rules/selector-no-vendor-prefix/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.messages = exports.ruleName = undefined;

exports.default = function (actual) {
  return function (root, result) {
    var validOptions = (0, _utils.validateOptions)(result, ruleName, { actual: actual });
    if (!validOptions) {
      return;
    }

    root.walkRules(function (rule) {
      var selector = rule.selector;

      // Check each pseudo-selector
      (0, _utils.styleSearch)({ source: selector, target: ":" }, function (match) {
        var pseudoSelector = /:{1,2}[a-z-]+\b/.exec(selector.slice(match.startIndex));
        if (pseudoSelector === null) {
          return;
        }
        if (_utils.isAutoprefixable.selector(pseudoSelector[0])) {
          (0, _utils.report)({
            message: messages.rejected(pseudoSelector[0]),
            node: rule,
            index: match.startIndex,
            result: result,
            ruleName: ruleName
          });
        }
      });
    });
  };
};

var _utils = require("../../utils");

var ruleName = exports.ruleName = "selector-no-vendor-prefix";

var messages = exports.messages = (0, _utils.ruleMessages)(ruleName, {
  rejected: function rejected(p) {
    return "Unexpected vendor-prefixed selector \"" + p + "\"";
  }
});
