// stylelint/dist/rules/selector-delimiter-space-after/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.selectorDelimiterSpaceChecker = selectorDelimiterSpaceChecker;

var _utils = require("../../utils");

var ruleName = "selector-delimiter-space-after";

exports.ruleName = ruleName;
var messages = (0, _utils.ruleMessages)(ruleName, {
  expectedAfter: function expectedAfter() {
    return "Expected single space after selector delimiter";
  },
  rejectedAfter: function rejectedAfter() {
    return "Unexpected space after selector delimiter";
  } });
exports.messages = messages;
/**
 * @param {"always"|"never"} expectation
 */

exports["default"] = function (expectation) {
  var checker = (0, _utils.whitespaceChecker)(" ", expectation, messages);
  return selectorDelimiterSpaceChecker(checker.after);
};

function selectorDelimiterSpaceChecker(checkLocation) {
  return function (css, result) {
    css.eachRule(function (rule) {
      var selector = rule.selector;
      (0, _utils.valueIndexOf)({ value: selector, char: "," }, function (index) {
        checkDelimiter(selector, index, rule);
      });
    });

    function checkDelimiter(source, index, node) {
      checkLocation(source, index, function (m) {
        return result.warn(m, { node: node });
      });
    }
  };
}
