// stylelint/dist/rules/declaration-semicolon-space-before/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _utils = require("../../utils");

var ruleName = "declaration-semicolon-space-before";

exports.ruleName = ruleName;
var messages = (0, _utils.ruleMessages)(ruleName, {
  expectedBefore: function expectedBefore() {
    return "Expected single space before \";\"";
  },
  rejectedBefore: function rejectedBefore() {
    return "Unexpected space before \";\"";
  },
  expectedBeforeSingleLine: function expectedBeforeSingleLine() {
    return "Expected single space before \";\" within single-line declaration block";
  },
  rejectedBeforeSingleLine: function rejectedBeforeSingleLine() {
    return "Unexpected space before \";\" within single-line declaration block";
  } });

exports.messages = messages;
/**
 * @param {"always"|"never"|"always-single-line"|"never-single-line"} expectation
 */

exports["default"] = function (expectation) {
  var check = (0, _utils.whitespaceChecker)(" ", expectation, messages);
  return function (css, result) {
    css.eachDecl(function (decl) {
      // Ignore last declaration if there's no trailing semicolon
      var parentRule = decl.parent;
      if (!parentRule.semicolon && parentRule.last === decl) {
        return;
      }

      var declString = decl.toString();
      check.before(declString, declString.length, function (m) {
        return result.warn(m, { node: decl });
      }, parentRule.toString().slice("{"));
    });
  };
};
