// stylelint/dist/rules/max-line-length/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.messages = exports.ruleName = undefined;

exports.default = function (length) {
  return function (root, result) {
    var validOptions = (0, _utils.validateOptions)(result, ruleName, {
      length: _lodash.isNumber
    });
    if (!validOptions) {
      return;
    }

    // Collapse all urls into something nice and short
    var rootString = root.source.input.css.replace(/url\(.*\)/g, "url()");

    // Check first line
    checkNewline({ endIndex: 0 });

    // Check subsequent lines
    (0, _utils.styleSearch)({ source: rootString, target: ["\n"], checkComments: true }, checkNewline);

    function checkNewline(match) {
      var nextNewlineIndex = rootString.indexOf("\n", match.endIndex);

      // Accommodate last line
      if (nextNewlineIndex === -1) nextNewlineIndex = rootString.length;

      if (nextNewlineIndex - match.endIndex <= length) {
        return;
      }

      // If there are no spaces besides initial (indent) spaces, ignore
      var lineString = rootString.slice(match.endIndex, nextNewlineIndex);
      if (lineString.replace(/^\s+/, "").indexOf(" ") === -1) {
        return;
      }

      (0, _utils.report)({
        message: messages.expected(length),
        node: root,
        index: nextNewlineIndex - 1,
        result: result,
        ruleName: ruleName
      });
    }
  };
};

var _lodash = require("lodash");

var _utils = require("../../utils");

var ruleName = exports.ruleName = "max-line-length";

var messages = exports.messages = (0, _utils.ruleMessages)(ruleName, {
  expected: function expected(l) {
    return "Expected line length equal to or less than " + l + " characters";
  }
});
