// stylelint/dist/rules/declaration-semicolon-newline-after/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _utils = require("../../utils");

var ruleName = "declaration-semicolon-newline-after";

exports.ruleName = ruleName;
var messages = (0, _utils.ruleMessages)(ruleName, {
  expectedAfter: function expectedAfter() {
    return "Expected newline after \";\"";
  },
  rejectedAfter: function rejectedAfter() {
    return "Unexpected newline after \";\"";
  },
  expectedAfterMultiLine: function expectedAfterMultiLine() {
    return "Expected newline after \";\" within multi-line declaration block";
  },
  rejectedAfterMultiLine: function rejectedAfterMultiLine() {
    return "Unexpected newline after \";\" within multi-line declaration block";
  } });

exports.messages = messages;
/**
 * @param {"always"|"never"|"always-multi-line"|"never-multi-line"} expectation
 */

exports["default"] = function (expectation) {
  var check = (0, _utils.whitespaceChecker)("\n", expectation, messages);
  return function (css, result) {
    css.eachDecl(function (decl) {
      // Ignore last declaration if there's no trailing semicolon
      var parentRule = decl.parent;
      if (!parentRule.semicolon && parentRule.last === decl) {
        return;
      }

      var nextDecl = decl.next();
      if (!nextDecl) {
        return;
      }
      check.after(nextDecl.toString(), -1, function (m) {
        return result.warn(m, { node: decl });
      }, parentRule.toString().slice("{"));
    });
  };
};
