// stylelint/dist/rules/declaration-comma-space-after/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.declarationCommaSpaceChecker = declarationCommaSpaceChecker;

var _utils = require("../../utils");

var ruleName = "declaration-comma-space-after";

exports.ruleName = ruleName;
var messages = (0, _utils.ruleMessages)(ruleName, {
  expectedAfter: function expectedAfter() {
    return "Expected single space after comma within a declaration value";
  },
  rejectedAfter: function rejectedAfter() {
    return "Unexpected space after comma within a declaration value";
  } });

exports.messages = messages;
/**
 * @param {"always"|"never"} expectation
 */

exports["default"] = function (expectation) {
  var checker = (0, _utils.whitespaceChecker)(" ", expectation, messages);
  return declarationCommaSpaceChecker(checker.after);
};

function declarationCommaSpaceChecker(checkLocation) {
  return function (css, result) {
    css.eachDecl(function (decl) {
      var value = decl.value;

      (0, _utils.valueIndexOf)({ value: value, char: ",", outsideFunction: true }, function (index) {
        checkComma(value, index, decl);
      });
    });

    function checkComma(source, index, node) {
      checkLocation(source, index, function (m) {
        return result.warn(m, { node: node });
      });
    }
  };
}
