// stylelint/dist/rules/rule-single-line-max-declarations/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _lodash = require("lodash");

var _utils = require("../../utils");

var ruleName = "rule-no-single-line";

exports.ruleName = ruleName;
var messages = (0, _utils.ruleMessages)(ruleName, {
  expected: function expected(quantity) {
    return "Expected a maximum of " + quantity + " declaration(s)";
  }
});

exports.messages = messages;

exports["default"] = function (quantity) {
  return function (root, result) {
    var validOptions = (0, _utils.validateOptions)(result, ruleName, {
      actual: quantity,
      possible: [_lodash.isNumber]
    });
    if (!validOptions) {
      return;
    }

    root.walkRules(function (rule) {

      if (!(0, _utils.isSingleLineString)(rule.toString())) {
        return;
      }

      var decls = rule.nodes.filter(function (node) {
        return node.type === "decl";
      });

      if (decls.length <= quantity) {
        return;
      }

      (0, _utils.report)({
        message: messages.expected(quantity),
        node: rule,
        result: result,
        ruleName: ruleName
      });
    });
  };
};
