// stylelint/dist/rules/nesting-block-opening-brace-space-before/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _utils = require("../../utils");

var _nestingBlockOpeningBraceNewlineBefore = require("../nesting-block-opening-brace-newline-before");

var ruleName = "nesting-block-opening-brace-space-before";

exports.ruleName = ruleName;
var messages = (0, _utils.ruleMessages)(ruleName, {
  expectedBefore: function expectedBefore() {
    return "Expected single space before \"{\"";
  },
  rejectedBefore: function rejectedBefore() {
    return "Unexpected whitespace before \"{\"";
  },
  expectedBeforeSingleLine: function expectedBeforeSingleLine() {
    return "Expected single space before \"{\" of a single-line nesting block";
  },
  rejectedBeforeSingleLine: function rejectedBeforeSingleLine() {
    return "Unexpected whitespace before \"{\" of a single-line nesting block";
  },
  expectedBeforeMultiLine: function expectedBeforeMultiLine() {
    return "Expected single space before \"{\" of a multi-line nesting block";
  },
  rejectedBeforeMultiLine: function rejectedBeforeMultiLine() {
    return "Unexpected whitespace before \"{\" of a multi-line nesting block";
  }
});

exports.messages = messages;

exports["default"] = function (expectation) {
  return function (root, result) {
    var validOptions = (0, _utils.validateOptions)(result, ruleName, {
      actual: expectation,
      possible: ["always", "never", "always-single-line", "never-single-line", "always-multi-line", "never-multi-line"]
    });
    if (!validOptions) {
      return;
    }

    (0, _nestingBlockOpeningBraceNewlineBefore.checkNestingBlockOpeningBraceBefore)({
      root: root,
      result: result,
      locationChecker: (0, _utils.whitespaceChecker)("space", expectation, messages),
      checkedRuleName: ruleName
    });
  };
};
