// stylelint/dist/rules/declaration-semicolon-space-after/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _utils = require("../../utils");

var ruleName = "declaration-semicolon-space-after";

exports.ruleName = ruleName;
var messages = (0, _utils.ruleMessages)(ruleName, {
  expectedAfter: function expectedAfter() {
    return "Expected single space after \";\"";
  },
  rejectedAfter: function rejectedAfter() {
    return "Unexpected space after \";\"";
  },
  expectedAfterSingleLine: function expectedAfterSingleLine() {
    return "Expected single space after \";\" within single-line declaration block";
  },
  rejectedAfterSingleLine: function rejectedAfterSingleLine() {
    return "Unexpected space after \";\" within single-line declaration block";
  } });

exports.messages = messages;
/**
 * @param {"always"|"never"|"always-single-line"|"never-single-line"} expectation
 */

exports["default"] = function (expectation) {
  var check = (0, _utils.whitespaceChecker)(" ", expectation, messages);
  return function (css, result) {
    css.eachDecl(function (decl) {
      // Ignore last declaration if there's no trailing semicolon
      var parentRule = decl.parent;
      if (!parentRule.semicolon && parentRule.last === decl) {
        return;
      }

      var nextDecl = decl.next();
      if (!nextDecl) {
        return;
      }
      check.after(nextDecl.toString(), -1, function (m) {
        return result.warn(m, { node: decl });
      }, parentRule.toString().slice("{"));
    });
  };
};
