// stylelint/dist/rules/no-multiple-empty-lines/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _utils = require("../../utils");

var ruleName = "no-multiple-empty-lines";

exports.ruleName = ruleName;
var messages = (0, _utils.ruleMessages)(ruleName, {
  rejected: "Unexpected empty line"
});

exports.messages = messages;

exports["default"] = function (actual) {
  return function (root, result) {
    var validOptions = (0, _utils.validateOptions)(result, ruleName, { actual: actual });
    if (!validOptions) {
      return;
    }

    var rootString = root.source.input.css;
    (0, _utils.styleSearch)({ source: rootString, target: "\n", checkComments: true }, function (match) {
      if (rootString.substr(match.startIndex + 1, 2) === "\n\n" || rootString.substr(match.startIndex + 1, 4) === "\r\n\r\n") {
        (0, _utils.report)({
          message: messages.rejected,
          node: root,
          index: match.startIndex,
          result: result,
          ruleName: ruleName
        });
      }
    });
  };
};
